docs/ccoctl.md

```diff
@@ -215,3 +215,124 @@ This command will delete the service id from the IBM Cloud
+   
+   ```
+
+**step 2&3 are only needed when preparing for upgrading clusters with manually maintained credentials. When doing a fresh installation please skip step 2&3**
+
+2. For each CredentialsRequest CR in the release image, ensure that a namespace that matches the text in the spec.secretRef.namespace field exists in the cluster. This field is where the generated secrets that hold the credentials configuration are stored.
+
+   Sample Alibaba Cloud CredentialsRequest object
+
+   ```yaml
+   apiVersion: cloudcredential.openshift.io/v1
+   kind: CredentialsRequest
+   metadata:
+     name: cloud-credential-operator-ram-ro
+     namespace: openshift-cloud-credential-operator
+   spec:
+     providerSpec:
+       apiVersion: cloudcredential.openshift.io/v1
+       kind: AlibabaCloudProviderSpec
+       statementEntries:
+       - action:
+         - ecs:CopySnapshot
+         - ecs:DeleteDisk
+         - ecs:DescribeInstanceAttribute
+         - ecs:DescribeInstances
+         effect: Allow
+         resource: '*'
+     secretRef:
+       namespace: cloud-credential-operator-ram-ro-creds
+       name: openshift-cloud-credential-operator
+   ```
+
+ 3. For any `CredentialsRequest` CR for which the cluster does not already have a namespace with the name specified in `spec.secretRef.namespace`, create the namespace:
+
+   ```bash
+   $ oc create namespace <component_namespace>
+   ```
+
+4. Use the `ccoctl` tool to process all `CredentialsRequest` objects in the `credrequests` directory:
+
+   ```bash
+   $ ccoctl alibabacloud create-ram-users --name <name> --region=<region> --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests --output-dir=xxxxxx
+   ```
+
+    where:
+
+   - `name` is the name used to tag any cloud resources that are created for tracking. 
+   - `region` is the Alibaba Cloud region in which cloud resources will be created.
+   - `credentials-requests-dir` is the directory containing files of component CredentialsRequests.
+   - `output-dir`/manifests is the directory containing files of component credentials secret.
+
+5. Prepare to run the OpenShift Container Platform installer:
+
+   a. Create the install-config.yaml file:
+   ```bash
+   $ openshift-install create install-config --dir ./path/to/installation/dir
+   ```
+   b. Configure the cluster to install with the CCO in manual mode:
+
+   ```bash
+   $ echo "credentialsMode: Manual" >> ./path/to/installation/dir/install-config.yaml
+   ```
+   
+   c. Create install manifests:
+
+   ```bash
+   $ openshift-install create manifests --dir ./path/to/installation/dir
+   ```
+
+   d. Copy the generated credential files to the target manifests directory:
+
+   ```bash
+   $ cp <output_dir>/manifests/*credentials.yaml ./path/to/installation/dir/manifests/
+   ```
+6. To delete resources created by ccoctl, run
```

Comment:
Should the user always delete after installing? Or is this a cleanup step for uninstalling (I am asking based on other provider docs)?

Reply:
These keys will most likely live until the cluster is uninstalled. The keys are being used by the cluster's internal components that require API access. They MUST not be deleted after installation or the cluster will not continue to function properly.

Reply:
Ok, so a cleanup after uninstall step similar to other platforms then. Thanks!